### DISCUSSION TITLE
Cusam CS-IRC-1 IR Controller: documented as tasmotizable, but how?

### DISCUSSION THREAD

**Discussion by ruedli** [arendst/Tasmota#14228] [Q&A] [answered]
I followed the aliexpress link from the "supported devices" tasmota page and ordered this one documented to support "USB to Serial" flashing method.

Can the IR blaster I received with an WBR3 module and non-exposed RXD/TXD pins be tasmotized at all?

I got a model CS-SSRC-IRC (description said CS-IRQ-1) that lacks TXD and RXD pins and seems to have a WBR3 module. I looks like this 
![image](https://user-images.githubusercontent.com/5008440/147850713-fad60057-13e2-4f9e-91b0-9e046b1c48dc.png)

As you can see there are no simple RXD and TXD pins exposed. This is the picture of a guide I found online for flashing an IRC module, which requires 2 more pins (and a different module?).

![image](https://user-images.githubusercontent.com/5008440/147850756-035e9334-9bd5-4c9d-976b-8f2426d25d06.png)

**Comment by barbudor** [comment, marked as answer]
Unfortunately WBR3 is not based on an ESP CPU, so not directly tasmotazible.
The good news is that the module has a compatible pinout with ESP12F standard module, or even the newer version ESP32-C3-12F which runs an ESP32c3

Need a bit a soldering skills but many have already done similar tweak known as Frankesteinization or Brain Swap

**Reply by ruedli** [reply]
Thanks, that saves a lot of trial and error!

So Frankensteining using this:
![image](https://user-images.githubusercontent.com/5008440/147851918-fcb3c41b-c3ea-4b70-bc14-eaeafbd20d27.png)

should work. It also has pins on the short edge, I guess these are not used (IO6 & IO7)?

TX RX need to be connected for flashing lower left of the antenna, like this?

![image](https://user-images.githubusercontent.com/5008440/147852012-db9e87b0-c4de-4190-85e8-7a282c7d0504.png)

**Reply by barbudor** [reply]
Beware that the pinout is as seen from the other side of the module, not the side with the metal cap
And you need the version with PCB antenna. The one selected in your screenshot has a connector for external antenna.

For flashing you will need TX, RX, GND and hold Gpio9 to GND.
Gpio9 is equivalent for C3 as gpio0 for esp8266 and conveniently located as the same location on the ESP12F module
You need to flash 'tasmota32c3'
Hope this helps

**Reply by barbudor** [reply]
And better use Tasmota web flasher (https://tasmota.github.io/install/) and look for 'Tasmota32c3 4mb' in the drop down

**Comment by ruedli** [comment]
Thnx for your detailed instructions on flashing, should be a piece of cake. 

I canceled the order for the antenna, as also I was not sure on the difference between C3-12F(out 4M) and C3-12f(in 4M), however, I believe it _has_ a PCB antenna, my screenshot was zoomed in and the antenna was only halve in it. This is the full screen shot, the antenna connector is optional and I assume the PCB antenna is hooked up by default:
![image](https://user-images.githubusercontent.com/5008440/147856244-90f5bc97-bb10-4d3b-9206-6e5a03cd4a8b.png)

Manual of the module is here: https://docs.ai-thinker.com/_media/esp32/docs/esp-c3-12f_specification.pdf

**Reply by barbudor** [reply]
Indeed, I thought it hasn't the PCB antenna because it was shorter 🤣 

"in" and "out" is not clear but it may refer to flash as listed in section 3:
> ESP-C3-12F have two FLASH， including external FLASH and build-in FLASH

So I would expect "in" to be better

**Reply by ruedli** [reply]
Exactly, I think we figured it out, the specification mentions that flash memory can be hooked up "internally" or "externally". Given that in/out is with the spec of the amount of memory, I agree that it refers to that: flash internal / external.

**Comment by wernerhp** [comment]
Can flash ESPHome using LibreTiny
https://docs.libretiny.eu/docs/status/supported/#tuya-inc

**Comment by synman** [comment]
apologies for practicing necromany... was a solution ever found for this?  I have a similar IR Blaster with the same question.  Am good using something other than Tasmota... 

![7cdd3bb95e092b9ad2e1007444f8b5dee94fda72aff2f61e37ef2efee7417b48](https://github.com/arendst/Tasmota/assets/1299716/a3d3ddd3-9604-48be-8658-4a6c1163cf53)

![a5e9cf58a706943b76ae46b1086d30160229f7c1714a7e222ed36f60e3f1dde7](https://github.com/arendst/Tasmota/assets/1299716/2956b352-4be4-452c-8516-863ba9f15fd6)

**Reply by mrtati** [reply]
@synman: Sorry to ask, but where did you get this ir blaster model near end of 2023?
It is very similar to what I have, BlitzWolf BW-RC02, as it has LEDs on the rim. It is sold out everywhere and I would like to buy a few more. I am looking for an OEM variant branded under a different name...

Btw, I managed to change it to Tasmota by replacing the wbr3 with esp12f. The RGB leds are also working.

**Comment by synman** [comment]
@wernerhp referenced ESPHome and LibreTiny but I see this controller on its unsupported list.

**Reply by wernerhp** [reply]
Oof! My bad, I saw it listed and didn't pay attention to the heading.

**Reply by synman** [reply]
> Oof! My bad, I saw it listed and didn't pay attention to the heading.

I've been digging a bit today and there is a footnote about using a generic 8720C template.

I was able to get a binary built in esp home but getting this controller to go into download mode is alluding to me.

**Reply by wernerhp** [reply]
Yeah  they can be tricky. Need the right Flasher that can supply enough power and then sometimes need some me tricks to get it in.  You need som persistence with this hobby 😄

**Reply by synman** [reply]
well I gave up after a bit of dorking around.  I've decided I'm just going to build my own IR blaster.   Should've done this to begin with as I'll be able to make 5 of them for the $20 I spent for this one.

**Reply by sfromis** [reply]
Replacing the WBR3 with a "Tasmota-compatible" module, as this thread is about, should not be a big job.

**Reply by synman** [reply]
Sure if you have eagle eyes and steady hands.

Meanwhile mere mortals rely on things like a 3d printed jig and pogo pins :)
